// SPDX-FileCopyrightText: 2025 Carnegie Mellon University
// SPDX-License-Identifier: GPL-2.0-only

syntax = "proto3";

package edu.cmu.cs.hawk;

// Object Identifier
// Opaque to everyone except for the specific retriever implementation.
message ObjectId {
  string oid = 1;
}

// "typed" binary object content
// media_type indicates how to interpret the binary data.
message HawkObject {
  string media_type = 1;
  bytes content = 2;
}

// Area/region of an image, the xywh coordinates are normalized relative to
// the image size so they don't change when the image is resized.
// full image is represented as [center_x=0.5, center_y=0.5, width=1.0, height=1.0]
message Region {
  float center_x = 1;
  float center_y = 2;
  float width = 3;
  float height = 4;
}

// Detected object
// can also be used as classification (coords undefined/cover the whole image)
message Detection {
  string class_name = 1;
  optional float confidence = 2;  // default: 1.0
  optional Region coords = 3;     // default: {0.5, 0.5, 1.0, 1.0}
}
